[1.20.x] Optimise `ForgeRegistry#validateContent`

PaintNinja opened the pull request:
I remember someone pointing out that the same method is looked up with `ObfuscationReflectionHelper` and called reflectively on every call to `validateContent()`.

This PR fixes that by ~~instead using a `MethodHandle` constant and invoking that if available~~ removing it as it's been broken since 1.17.

PaintNinja committed:
Optimise `ForgeRegistry#validateContent`

PaintNinja committed:
Clarify a code comment

PaintNinja committed:
Revert "Clarify a code comment"

This reverts commit 7c27439ee888f3b7922b586b346c819aa5ffb502.

That was a sleepy mistake

PaintNinja commented:
Leaving as a draft until I'm less sleepy to make sure this is right.

Jonathing commented on src/main/java/net/minecraftforge/registries/ForgeRegistry.java:
```diff
@@ -54,8 +56,12 @@
-    private static Marker REGISTRYDUMP = MarkerManager.getMarker("REGISTRYDUMP");
-    private static Logger LOGGER = LogManager.getLogger();
+    private static final Marker REGISTRYDUMP = MarkerManager.getMarker("REGISTRYDUMP");
+    private static final Logger LOGGER = LogManager.getLogger();
```
Try to avoid unnecessary changes like these in scoped PRs. Yes, they're good, but I think that if it isn't part of the PR, it should be saved for later.

PaintNinja replied:
Bit nit-picky tbh... It's nearby and obvious, with such a small change not really warranting the effort of making a new branch, setting up Forge again, creating a PR, etc...

Jonathing replied:
Yeah no reason for you to change it back, that's why I mentioned it but still gave the thumbs up.

Jonathing reviewed (approved):
It's a little difficult to read, but I understand what's happening here. All-in-all, looks good. I personally tend to prefer to put comments explaining hacky shit.

Jonathing reviewed (changes requested):
Oops

embeddedt commented:
FWIW, refactoring to use a MethodHandle is overkill. The cost was in looking up the `Method` every time, not the actual invocation. Saving the returned `Method` from `findMethod` would be enough.

PaintNinja committed:
Remove broken reflection

This hasn't worked since 1.17, so no point caching the reflected method as it'll always be null...

PaintNinja commented:
> FWIW, refactoring to use a MethodHandle is overkill. The cost was in looking up the `Method` every time, not the actual invocation. Saving the returned `Method` from `findMethod` would be enough.

Thanks for the suggestion, however after testing (logging the exceptions) I've discovered it hasn't worked since 1.17 due to JPMS restrictions and ModLauncher limitations, so I've removed it altogether:

`Caused by: java.lang.IllegalAccessError: class net.minecraftforge.registries.ForgeRegistry tried to access private method 'void java.util.BitSet.trimToSize()' (net.minecraftforge.registries.ForgeRegistry is in module forge@47.1.50-1-20-optimise-forgeregistry-validatecontent of loader 'TRANSFORMER' @29b40b3; java.util.BitSet is in module java.base of loader 'bootstrap')`

PaintNinja committed:
Remove unnecessary imports
